core_impl/src/main/java/io/opencensus/implcore/tags/TaggerImpl.java

```diff
@@ -52,16 +62,19 @@ public TagContextBuilder currentBuilder() {
   }
 
   @Override
-  public TagContextBuilderImpl toBuilder(TagContext tags) {
-    return toTagContextBuilderImpl(tags);
+  public TagContextBuilder toBuilder(TagContext tags) {
+    return toTagContextBuilderImpl(tags, state.get());
   }
 
   @Override
   public Scope withTagContext(TagContext tags) {
-    return CurrentTagContextUtils.withTagContext(toTagContextImpl(tags));
+    return CurrentTagContextUtils.withTagContext(toTagContextImpl(tags, state.get()));
   }
 
-  private static TagContextImpl toTagContextImpl(TagContext tags) {
+  private static TagContextImpl toTagContextImpl(TagContext tags, TaggingState state) {
```

Comment:
Also I think we discussed to not change this here and do it only when we need to do it (record usage maybe when add tags).